components/frontend/src/source/SourceParameters.js

```diff
@@ -1,51 +1,87 @@
-                help={all_parameters[parameter_key].help}
-                requiredPermissions={[EDIT_REPORT_PERMISSION]}
-                placeholder={all_parameters[parameter_key].placeholder || ""}
-                required={all_parameters[parameter_key].mandatory}
-                warning={changed_param_keys?.indexOf(parameter_key) !== -1}
-                reload={reload}
-                index={index}
-            />
-        </Grid.Column>
-    )
-    );
-    return (<>{parameters}</>)
+    const allParameters = dataModel.sources[source.type].parameters;
+    const parameterLayout = dataModel.sources[source.type].parameter_layout ?? DEFAULT_LAYOUT;
+    const groupedParameters = collectGroupedParameters(parameterLayout)
+    const remainingParameters = collectRemainingParameters(allParameters, groupedParameters)
+    const groups = Object.values(parameterLayout).map((parameterGroup) => {
+        const parameterKeys = applicableParameters(allParameters, remainingParameters, parameterGroup, metric)
+        if (parameterKeys.length === 0) {
+            return null
+        }
+        const parameters = parameterKeys.map((parameterKey, index) => (
+            <div key={parameterKey} style={{ paddingTop: '10px' }}>
+                <SourceParameter
+                    report={report}
+                    source={source}
+                    source_uuid={source_uuid}
+                    source_type_name={dataModel.sources[source.type].name}
+                    parameter_key={parameterKey}
+                    parameter_type={allParameters[parameterKey].type}
+                    parameter_name={allParameters[parameterKey].name}
+                    parameter_short_name={allParameters[parameterKey].short_name}
+                    parameter_unit={allParameters[parameterKey].unit || metricUnit}
+                    parameter_min={allParameters[parameterKey].min_value || null}
+                    parameter_max={allParameters[parameterKey].max_value || null}
+                    parameter_value={source.parameters?.[parameterKey] ?
+                        source.parameters[parameterKey] : allParameters[parameterKey].default_value}
+                    parameter_values={allParameters[parameterKey].values || []}
+                    help_url={allParameters[parameterKey].help_url}
+                    help={allParameters[parameterKey].help}
+                    requiredPermissions={[EDIT_REPORT_PERMISSION]}
+                    placeholder={allParameters[parameterKey].placeholder || ""}
+                    required={allParameters[parameterKey].mandatory}
+                    warning={changed_param_keys?.indexOf(parameterKey) !== -1}
+                    reload={reload}
+                    index={index}
+                />
+            </div>
+        ));
+        return (
+            <Grid.Column key={parameterGroup.name}>
+                <Segment>
+                    <Header as="h5" color="grey">{parameterGroup.name}</Header>
```

Comment:
Does this `grey` also work in dark mode?

Reply:
Yup, works fine.